src/Elasticsearch/TokenQueryBuilder.php

```diff
@@ -113,27 +117,48 @@ private function matchQuery(Field $field, string $token, SearchFieldConfig $conf
             $searchField = $config->getField() . '.search';
             $operator = $config->isAndLogic() ? 'and' : 'or';
 
-            $tokenCount = \count(\explode(' ', $token));
+            $tokens = \explode(' ', $token);
+            $tokenCount = \count($tokens);
 
+            // apply exact match
+            $queries[] = $tokenCount === 1
+                ? new TermQuery($config->getField(), $token, ['boost' => 1])
+                : new TermsQuery($config->getField(), $tokens, ['boost' => 1]);
+
+            $lastWord = $tokens[$tokenCount - 1];
```

Comment:
```suggestion
            $lastWord = $tokens[array_key_last($tokens)];
```

or if we use the `symfony/polyfill-php85`, we can use

```suggestion
            $lastWord = array_last($tokens);
```